### PR TITLE
[USH-919] Add pull request template for platform client (mzima)

### LINKED ISSUE #625: [USH-919] Add pull request template for platform client (mzima)
#### Issue description and expected output/solution

Prevent "from scratch" writing of pull request descriptions when contributors are submitting pull requests - add a pull request template for [platform client (mzima)](https://github.com/ushahidi/platform-client-mzima). How to add a pull request template?

* Take a clue from the [legacy platform client's .github folder](https://github.com/ushahidi/platform-client)
* Also see [Github documentation](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository)

<sub>From [SyncLinear.com](https://synclinear.com) | [USH-919](https://linear.app/ushahidi/issue/USH-919/add-pull-request-template-for-platform-client-mzima)</sub>

### PR DESCRIPTION
**Pull request by shakirandagire**
@Ifycode Could you help me review this issue 
- Fixes #625

### CODE REVIEW

**Comment by Ifycode** [comment]
I love that your PR title is clean, and same as the issue title. And it's linked to the issue you are fixing. I have some change requests which I will be dropping shortly @shakirandagire

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## Description
+
+Please include the summary of the changes made and any relevant context for this ticket.
+
+## Type of change
```
Comment (Ifycode) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Just keep the PR template simple just like the example sited in the issue ticket.
Most issues don't require this much things you are adding here.
If the contributor feels they need all kind of info due to the nature of the issue they are fixing, they can decide to add it themselves. 

That said, we can improve the markdown formatting of the example sited in issue ticket. And we can borrow something from what you added. Just not in the bulky format in which you supplied it.

See my next comment below to see the change to make and the markdown formatting.

Comment (Ifycode) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: #### This pull request makes the following changes
- Fixes ushahidi/platform-client-mzima#replace-this-text-with-issue-ticket-number

#### Pull request change details
What fixes does this pull request contain? - Add them here using ordered or unordered list.

#### Testing checklist
What reviewers or QA should check for while reviewing this pull request:
- [ ] Add checklist item 1 here
- [ ] Add cheklist item 2 here... Continue adding as much checklist items as you would like
- [ ] I certify that I ran my checklist

Ping @ushahidi/platform-client-mzima

**Review by Ifycode — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (shakirandagire) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: @Ifycode thanks for the suggestions, let me implement them.

**Comment by shakirandagire** [comment]
@Ifycode I have implemented the feedback left on the PR.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+## This pull request makes the following changes
+
```
Comment (Ifycode) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:2]: Use heading 4 `####` and not heading 2 `##` for all your headings

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+What fixes does this pull request contain? - Add them here using ordered or unordered list.
+
+## Testing checklist
+
+What reviewers or QA should check for while reviewing this pull request:
+
+[ ] Add checklist item 1 here
+[ ] Add cheklist item 2 here... Continue adding as much checklist items as you would like
+[ ] I certify that I ran my checklist
```
Comment (Ifycode) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: This is what your submission looks like in the screenshot below. You can see that the checkboxes are not well formatted @shakirandagire 

#

Screenshot 2023-10-23 at 12 08 56

**Review by Ifycode — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 3 above)
Comment (Ifycode) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Make reference my previous comment and also fix your formatting for unordered list too (i.e. the "Fixes ushahidi/platform-client-mzima#replace-this-text-with-issue-ticket-number" part): https://github.com/ushahidi/platform-client-mzima/pull/657#discussion_r1368254806

In summary, use the exact markdown formatting in that comment for your submission @shakirandagire

Comment (shakirandagire) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: @Ifycode I had primarily picked the text from the comment and not the formatting. On it now

Comment (shakirandagire) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: @Ifycode I have rectified the formatting of the PR template and it is currently like below
Screenshot 2023-10-23 at 14 39 51

**Review by Ifycode — APPROVED** [review]
Approved 🎉
Nicely done. Thank you for your contribution.

**Pull request merged**